GCEWindowsAgent/main.go

```diff
@@ -232,19 +236,18 @@ func main() {
 	}
 	logger.Init(ctx, opts)
 
-	var action string
-	if len(os.Args) < 2 {
-		action = "run"
-	} else {
-		action = os.Args[1]
-	}
+	action := flag.String("action", "run", "the action to run, one of run, install, remove, start, stop, or noservice")
+	testMode = flag.Bool("test_mode", false, "when in test mode, no calls are made to the metadata server during snapshots to retrieve config values")
```

Comment:
not sure we need to document and expose this sort of feature. i also think you should be more targeted - not a global test mode but a specific sort of test mode